slsim/Sources/source.py

```diff
@@ -125,6 +137,80 @@ def kwargs_variability_extracted(self):
                             "MJD": times,
                             name: magnitudes,
                         }
+
+            # Check if AGN model is used
+            elif any(
+                element in kwargs_variability_list_agn
+                for element in list(self.kwargs_variability)
+            ):
+
+                z = self.source_dict["z"]
+                if self.cosmo is None:
+                    raise ValueError(
+                        "Cosmology cannot be None for AGN class. Please"
+                        "provide a suitable astropy cosmology."
+                    )
+
+                else:
+                    # Get the dict object from the astropy.table.column
+                    agn_kwarg_dict = self.source_dict["kwargs_agn_model"].data[0][0]
+
+                    # Populate "None" for optional keys related to drawing random AGN
+                    random_kwargs = {"random_seed": None, "input_agn_bounds_dict": None}
+                    for opt_key in random_kwargs.keys():
+                        if opt_key in agn_kwarg_dict.keys():
+                            random_kwargs[opt_key] = agn_kwarg_dict[opt_key]
+                            del agn_kwarg_dict[opt_key]
+
+                    # Create the agn object
+                    self.agn_class = agn.RandomAgn(
+                        self.source_dict["i_band_mag"],
+                        self.source_dict["z"],
+                        cosmo=self.cosmo,
+                        random_seed=random_kwargs["random_seed"],
+                        input_agn_bounds_dict=random_kwargs["input_agn_bounds_dict"],
+                        **agn_kwarg_dict
+                    )
+
+                    # Get mean mags across LSST filters
+                    # Fix to eventually give the user power to choose other surveys
+                    mean_magnitudes = self.agn_class.get_mean_mags("lsst")
+
+                    # These are lsst filters, hard coded in for now. Fix later.
+                    filters = ["u", "g", "r", "i", "z", "y"]
```

Comment:
you can demand to pass these filters through kwargs_variability and can be used whenever needed. We have done this in supernovae case.